basis/ui/backend/cocoa/views/views.factor

```diff
@@ -445,38 +463,58 @@ PRIVATE>
                     text -> string CF>string str!               
                 ] if
                 window world-focus :> gadget
-                gadget support-input-methods? [
-                    gadget preedit? [
-                        gadget [ remove-preedit-text ] [ remove-preedit-info ] bi
-                    ] when
-                    str gadget user-input* drop                    
-                    f gadget preedit-selection-mode?<<
-                ] [ 
-                    str window user-input
-                ] if
+                gadget [
```

Comment:
previously we didn't have this wrapped in ``gadget [ ... ] when``, does it need that now?

Reply:
As you say, this new extra guard should not be needed. However, when I was working on better NSTextInputClient support for this code, the word that was supposed to be dispatched to a gadget was dispatched to "f", Mac Factor downed.

The problem may be a transitional problem because the backend has been changed since it will not be replicated, but I was afraid it would be to someone. So, I added that guard to all methods of NSTextInputClient protocol.